tools/mock-okta/util.js

```diff
@@ -203,10 +204,14 @@ util.mapRequestToCache = (req) => {
     data.responseMode = query.response_mode;
     data.state = query.state;
     data.nonce = query.nonce;
-    req.url = req.url
-      .replace(`state=${query.state}`, 'state=STATE')
-      .replace(`nonce=${query.nonce}`, 'nonce=NONCE');
 
+    // If playing back the tapes, replace the state in req.url with a known state
+    if (!record) {
+      req.url = req.url
```

Comment:
Why are we assigning it to itself?

Reply:
We're not assigning it to itself. It's actually replacing the state in the URL.
      
req.url = req.url.replace(`state=${query.state}`, `state=${config.state}`);

Reply:
Hah. My bad. Not sure how I missed that!